apple/Sources/FerrostarSwiftUI/Views/InstructionsView.swift

```diff
@@ -57,8 +57,14 @@ public struct InstructionsView: View {
         _sizeWhenNotExpanded = sizeWhenNotExpanded
     }
 
+    /// The visual instructions *after* the one for the current maneuver,
+    /// paired with the route step.
+    ///
+    /// Note that in the case of steps with multiple instructions,
+    /// only the first is returned in this list.
+    /// This is sufficient for display of the upcoming maneuver list.
     private var nextVisualInstructions: [(VisualInstruction, RouteStep)] {
-        guard let remainingSteps, !remainingSteps.isEmpty else {
```

Comment:
I prefer your change because it's clearer, but I don't see how the previous version was a bug. 

If remaining steps had one thing in it, then `remainingSteps[1...] == []` so the result is the same, right?

Or am I missing something?

Reply:
So I went off to code up a Swift Fiddle, because OBVIOUSLY Swift is zero-indexed, and accessing index 1 of an array of length 1 is out of bounds access... 

> You can subscript an array with any integer from zero up to, but not including, the count of the array. Using a negative number or an index equal to or greater than count triggers a runtime error.

But' here's a fiddle that proves your original code actually returns empty array as you claim: https://swiftfiddle.com/aw3i6znzvjcjzi3pdoefmu23sy. I've been writing Swift since the 0.x days and I have to say, this is the most surprised I've been in a very long time! 🤯

Scouring the Apple docs, I couldn't figure out why this works as it does, but your code was indeed correct 😂 If you've got any links explaining why this is the way it is, I'd be grateful to learn!

Reply:
```swift 
let foo = ["a"]
print("foo[0] => \(foo[0])") // "a"
// print("foo[1] => \(foo[1])") // Fatal error: Index out of range

print("foo[0...] => \(foo[0...])") // ["a"]
print("foo[1...] => \(foo[1...])") // []

// And my favorite - while this works (and seems like it should):
print("foo[0...] => \(foo[0...])") // ["a"]
print("foo[0..0] => \(foo[0...0])") // ["a"]

// This blows up
print("foo[1...] => \(foo[1...])") // []
// print("foo[1..1] => \(foo[1...1])") // Fatal error: Array index is out of range

// I think this is a hint to the riddle
// print("foo[2...] => \(foo[2...])") // Swift/Range.swift:754: Fatal error: Range requires lowerBound <= upperBound

// Note the error! 
// It's obvious where the lower bound is coming from (2), 
// but where is the upper bound coming from that is greater than 2?

func example() {
  // So my guess is that accessing a RandomAccessCollection with PartialRangeFrom, 
  // converts the PartialRangeFrom to a Range (not a ClosedRange!), using the Array.length as the `upperBound` of the Range

  do {
    let rangeFrom: PartialRangeFrom = 2...
    let arrayLength = 1
    // this will panic when rangeFrom.start is greater than arrayLength - e.g. 2..<1
    let range: Range = rangeFrom.lowerBound..<arrayLength 
  }

  do {
    // but this will not panic, it'll just be empty:
    let rangeFrom: PartialRangeFrom = 1...
    let arrayLength = 1
    let range: Range = rangeFrom.lowerBound..<arrayLength 
    assert(range == 1..<1)
  }
}
```

Link to fiddle: https://swiftfiddle.com/l4jhe6atjjgkdnih6ot6rsyzgy